### DISCUSSION TITLE
What can I do to reduce my access latency?

### DISCUSSION THREAD

**Discussion by Zerolzj** [anacrolix/torrent#958] [General]
When processing piece requests across two peers, the sequence of 'request 1 - cancel 1 - request 2' takes over 200ms (I can't stand) because of `cancelRequest`, despite the network having sufficient capacity to manage numerous requests concurrently. I prefer not to cancel the initial request if it has already been dispatched. Can you recommend best practices to address this issue?
https://github.com/anacrolix/torrent/blob/f22cf7de8b38ebc2d58f8cf2f096f9265735e1ef/requesting.go#L323-L340

**Comment by anacrolix** [comment]
I think you'll find if there's capacity in the network, the peer will issue the piece data for request 1 before the cancel 1 arrives.

The default request strategy tries to minimize unnecessarily duplicated piece data. There used to be several modes built in, but I eventually collapsed them all into the current request strategy. It's definitely conservative if your peers are very trustworthy and low latency.
